framework/src/main/java/org/apache/drill/test/framework/ColumnList.java

```diff
@@ -147,17 +154,23 @@ private boolean compare(ColumnList o1, ColumnList o2) {
       if (oneNull(list1.get(i), list2.get(i))) {
         return false;
       }
-      int type = (Integer) (types.get(i));
+      int type = types.get(i);
       try {
         switch (type) {
         case Types.FLOAT:
         case Types.REAL:
           float f1 = (Float) list1.get(i);
           float f2 = (Float) list2.get(i);
-          if ((f1 + f2) / 2 != 0) {
-            if (!(Math.abs((f1 - f2) / ((f1 + f2) / 2)) < 1.0E-6)) return false;
-          } else if (f1 != 0) {
-            return false;
+          if (f1 != f2) {
```

Comment:
You removed the check for a zero denominator.

Does this work if f1 is zero and f2 is negative?

Reply:
the prev 0 check was redundant ,but valid point id does need a zero denominator check